modules/dnn/src/ocl4dnn/src/ocl4dnn_pool.cpp

```diff
@@ -0,0 +1,202 @@
+            if (top_mask.empty() && mask_idx_.empty())
+            {
+                mask_idx_.create(1, count_, CV_32FC1);
+            }
+            ocl::Kernel oclk_max_pool_forward(CL_KERNEL_SELECT("max_pool_forward"),
+                                              cv::ocl::dnn::ocl4dnn_pooling_oclsrc);
+
+            argIdx = 0;
+            oclk_max_pool_forward.set(argIdx++, count_);
+            oclk_max_pool_forward.set(argIdx++, (cl_mem) bottom.handle(ACCESS_READ));
+            oclk_max_pool_forward.set(argIdx++, batch_size_);
+            oclk_max_pool_forward.set(argIdx++, channels_);
+            oclk_max_pool_forward.set(argIdx++, height_);
+            oclk_max_pool_forward.set(argIdx++, width_);
+            oclk_max_pool_forward.set(argIdx++, pooled_height_);
+            oclk_max_pool_forward.set(argIdx++, pooled_width_);
+            oclk_max_pool_forward.set(argIdx++, kernel_h_);
+            oclk_max_pool_forward.set(argIdx++, kernel_w_);
+            oclk_max_pool_forward.set(argIdx++, stride_h_);
+            oclk_max_pool_forward.set(argIdx++, stride_w_);
+            oclk_max_pool_forward.set(argIdx++, pad_h_);
+            oclk_max_pool_forward.set(argIdx++, pad_w_);
+            oclk_max_pool_forward.set(argIdx++, (cl_mem) top.handle(ACCESS_WRITE));
+            oclk_max_pool_forward.set(argIdx++, mask_idx_.empty() ? 0 : 1);
+            oclk_max_pool_forward.set(argIdx++, mask_idx_.empty() ? NULL :
+                                      (cl_mem) mask_idx_.handle(ACCESS_WRITE));
+            oclk_max_pool_forward.set(argIdx++, (cl_mem) top_mask.handle(ACCESS_WRITE));
+
+            oclk_max_pool_forward.run(1, global, local, false);
+        }
+        break;
+    case LIBDNN_POOLING_METHOD_AVE:
+        {
+            ocl::Kernel oclk_ave_pool_forward(CL_KERNEL_SELECT("ave_pool_forward"),
+                                              cv::ocl::dnn::ocl4dnn_pooling_oclsrc);
+
+            argIdx = 0;
+            oclk_ave_pool_forward.set(argIdx++, count_);
+            oclk_ave_pool_forward.set(argIdx++, (cl_mem) bottom.handle(ACCESS_READ));
+            oclk_ave_pool_forward.set(argIdx++, batch_size_);
+            oclk_ave_pool_forward.set(argIdx++, channels_);
+            oclk_ave_pool_forward.set(argIdx++, height_);
+            oclk_ave_pool_forward.set(argIdx++, width_);
+            oclk_ave_pool_forward.set(argIdx++, pooled_height_);
+            oclk_ave_pool_forward.set(argIdx++, pooled_width_);
+            oclk_ave_pool_forward.set(argIdx++, kernel_h_);
+            oclk_ave_pool_forward.set(argIdx++, kernel_w_);
+            oclk_ave_pool_forward.set(argIdx++, stride_h_);
+            oclk_ave_pool_forward.set(argIdx++, stride_w_);
+            oclk_ave_pool_forward.set(argIdx++, pad_h_);
+            oclk_ave_pool_forward.set(argIdx++, pad_w_);
+            oclk_ave_pool_forward.set(argIdx++, (cl_mem) top.handle(ACCESS_WRITE));
+
+            oclk_ave_pool_forward.run(1, global, local, false);
```

Comment:
Error checks are mandatory for `kernel.run()` calls

Reply:
Fixed in the update patch.